openebs/pkg/volume/provision_v1alpha1.go

```diff
@@ -0,0 +1,201 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/external-storage/lib/controller"
+	"github.com/kubernetes-incubator/external-storage/lib/util"
+	"github.com/kubernetes-incubator/external-storage/openebs/pkg/apis/v1alpha1"
+	mApiv1 "github.com/kubernetes-incubator/external-storage/openebs/pkg/v1"
+	mayav1 "github.com/kubernetes-incubator/external-storage/openebs/types/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type openEBSProvisionerV1alpha1 struct {
+	// Maya-API Server URI running in the cluster
+	mapiURI string
+
+	// Identity of this openEBSProvisioner, set to node's name. Used to identify
+	// "this" provisioner's PVs.
+	identity string
+}
+
+// NewOpenEBSProvisioner creates a new openebs provisioner
+func NewOpenEBSProvisionerV1alpha1(client kubernetes.Interface) controller.Provisioner {
+
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		glog.Errorf("ENV variable 'NODE_NAME' is not set")
+	}
+	var openebsObj mApiv1.OpenEBSVolume
+
+	//Get maya-apiserver IP address from cluster
+	addr, err := openebsObj.GetMayaClusterIP(client)
+
+	if err != nil {
+		glog.Errorf("Error getting maya-apiserver IP Address: %v", err)
+		return nil
+	}
+	mayaServiceURI := "http://" + addr + ":5656"
+
+	//Set maya-apiserver IP address along with default port
+	os.Setenv("MAPI_ADDR", mayaServiceURI)
+
+	return &openEBSProvisionerV1alpha1{
+		mapiURI:  mayaServiceURI,
+		identity: nodeName,
+	}
+}
+
+var _ controller.Provisioner = &openEBSProvisionerV1alpha1{}
+
+// Provision creates a storage asset and returns a PV object representing it.
+func (p *openEBSProvisionerV1alpha1) Provision(options controller.VolumeOptions) (*v1.PersistentVolume, error) {
+
+	//Issue a request to Maya API Server to create a volume
+	var volume mayav1.Volume
+	var openebsCASVol mApiv1.OpenEBSVolumeV1Alpha1
+	casVolume := v1alpha1.CASVolume{}
+
+	volSize := options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
+	casVolume.Spec.Capacity = volSize.String()
+
+	className := GetStorageClassName(options)
+
+	// creating a map b/c have to initialize the map using the make function before
+	// adding any elements to avoid nil map assignment error
+	mapLabels := make(map[string]string)
+
+	if className == nil {
+		glog.Errorf("Volume has no storage class specified")
+	} else {
+		mapLabels[string(v1alpha1.StorageClassKey)] = *className
+		glog.Infof("Using Storageclass: %s", *className)
+		casVolume.Labels = mapLabels
+	}
+	casVolume.Labels[string(v1alpha1.NamespaceKey)] = options.PVC.Namespace
```

Comment:
options.PVC.namespace may not be the only way. We might need to check openebs custom labels of PVC as well to get the namespace. Some users dont want the openebs pods to use the namespace of app pod which is also mandated to be set against the pvc spec. 
cc @kmova